client/src/components/BoardGameShowPage.js

```diff
@@ -28,17 +33,51 @@ const BoardGameShowPage = (props) => {
 		}
 	}
 
+	const handleDelete = async () => {
+		try {
+			const response = await fetch(`/api/v1/board-games/${id}`, {
+				method: "DELETE",
+				headers: new Headers({
+					"Content-Type": "application/json"
+				})
+			})
+			if (!response.ok) {
+				if (response.status === 422) {
+					const newErrors = translateServerErrors(body.errors)
+					return setErrors(newErrors)
+				} else {
+					const errorMessage = `${response.status} (${response.statusText})`
+					const error = new Error(errorMessage)
+					throw(error)
+				}
+			} else {
+				props.history.push("/board-games")
+				setErrors([])
+			}
+		} catch(err) {
+			console.error(`Error in fetch: ${err.message}`)
+		}
+	}
+
 	useEffect(() => {
 		getBoardGame()
 	}, [])
 
+	useEffect(() => {
+		if (currentUserId === boardGame.userId || role === "admin") {
+			setShowButton(true)
+		}
+	})
+	console.log(boardGame.userId)
```

Comment:
slow...down...

- we have a console.log that definitely needs to be removed

- we dont need this second useEffect at all, nor do we need `showButton` state. Just use the same if statement you have here in place of the check you have on line 94. Effectively, have an `if` statement that checks for the userId and the admin role, and conditionally render the DeleteCurrentGame button